### PR TITLE
Engagement unit test

### LINKED ISSUE #1133: Add Engagement Unit Tests
Please use the following example at as a template:
https://github.com/DefectDojo/django-DefectDojo/blob/dev/tests/Product_unit_test_example.py

Implement the following tests:
- Add
- Edit
- Close
- Delete
- Import scan results
- Add new CI/CD Engagement

Master List and Overview: https://docs.google.com/document/d/1dSS7tWuQukAX3VWtzPAlOfFNoqLQh-mBHBfmsu2WlU0/edit?usp=sharing

### PR DESCRIPTION
**Pull request by dr3dd589**
fix #1133 

implemented the following tests:

- [x] Add
- [x] Edit
- [x] Close
- [x] Delete
- [x] Import scan results
- [x] Add new CI/CD Engagement

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant (DefectDojo's code isn't currently flake8 compliant, but we're trying to correct that.)
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [ ] Model changes should include the necessary migrations in the dojo/dd_migrations folder.
- [ ] Add applicable tests to the unit tests.

### CODE REVIEW

File: tests/Engagement_unit_test.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        driver.find_element_by_link_text('Delete Engagement').click()
+        driver.find_element_by_name('delete_name').click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Engagement and relationships removed.', EngagementTXT))
+
+    def test_new_ci_cd_engagement(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_link_text('Engagement Test').click()
+        driver.find_element_by_xpath("//a[@class='dropdown-toggle active']//span[@class='hidden-xs']").click()
```
Comment (Maffooch) [inline tests/Engagement_unit_test.py]: This line gave me issues. Commenting it allowed the test to run to success. Not sure it's necessary.

File: tests/Engagement_unit_test.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        self.assertTrue(re.search(r'Engagement added successfully.', EngagementTXT))
+
+    def tearDown(self):
+        self.driver.quit()
+        self.assertEqual([], self.verificationErrors)
+
+
+def suite():
+    suite = unittest.TestSuite()
+    suite.addTest(EngagementTest('create_new_product_to_test_engagement'))
```
Comment (Maffooch) [inline tests/Engagement_unit_test.py]: Be sure to clean up used resources. Deleting a product also deletes all engagements/findings/.etc.

File: tests/Engagement_unit_test.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        driver = self.driver
+        driver.get(self.base_url + "login")
+        driver.find_element_by_id("id_username").clear()
+        driver.find_element_by_id("id_username").send_keys(os.environ['DD_ADMIN_USER'])
+        driver.find_element_by_id("id_password").clear()
+        driver.find_element_by_id("id_password").send_keys(os.environ['DD_ADMIN_PASSWORD'])
+        driver.find_element_by_css_selector("button.btn.btn-success").click()
+        return driver
+
+    def create_new_product_to_test_engagement(self):
```
Comment (Maffooch) [inline tests/Engagement_unit_test.py]: ProperSam is working on a way to import this test from the Product test suite to reduce the amount of copy/paste code. coordinate with him to find a way to do that in a manner that is compatible with python3. Check his PRs for reference on what has already been attempted.

File: tests/Engagement_unit_test.py (hunk 1 above)
Comment (dr3dd589) [reply tests/Engagement_unit_test.py]: yes but that click on import result but I want to click on Done button to separate add engagement test.

File: tests/Engagement_unit_test.py (hunk 2 above)
Comment (dr3dd589) [reply tests/Engagement_unit_test.py]: yes, that's why I created a new product and test all engagement separately.

File: tests/Engagement_unit_test.py (hunk 3 above)
Comment (dr3dd589) [reply tests/Engagement_unit_test.py]: okk.

File: tests/Engagement_unit_test.py (hunk 1 above)
Comment (Maffooch) [reply tests/Engagement_unit_test.py]: This is due to this issue https://github.com/DefectDojo/django-DefectDojo/issues/1160. The version without the pretty UI doesn't have a drop down. Everything is plainly listed. Hold tight for now.

File: tests/Engagement_unit_test.py (hunk 3 above)
Comment (propersam) [reply tests/Engagement_unit_test.py]: oh!.
about that, I have already created a line of code using try and except to decide between executing python 2 or python 3 code for importing your create product from another module.
of course, you would have to tweak it to use create product code from your own product_type_test.py file

 check this commit https://github.com/DefectDojo/django-DefectDojo/pull/1166/commits/ab7b88ecfcd2fad5c147206e7b4c5b1023a43f41

Comment (propersam) [reply tests/Engagement_unit_test.py]: don't forget to modify ur build Test suit function  at the button..

scroll down in this earlier commit to see how I did mine..

https://github.com/DefectDojo/django-DefectDojo/pull/1167/commits/0516f8c66553c93ea68fbb229148c1e00029c57a

Comment (dr3dd589) [reply tests/Engagement_unit_test.py]: thanks, @propersam but I have not created any additional product in product type test so I can't just import add product function from another file (bcs i dont have :) ) the only way is we have to wait for either your pr will be merged or mine and then we can import add product function.

Comment (Maffooch) [reply tests/Engagement_unit_test.py]: Product unit test has been merged. Probably be easiest to just copy/paste from the Endpoint unit test to import the tests. Best of luck!

**Comment by dr3dd589** [comment]
@Maffooch done!

**Comment by Maffooch** [comment]
@dr3dd589 Don't forget the delete product as well.

**Comment by dr3dd589** [comment]
ohh! sorry about that. :)

**Pull request merged**